src/components/admin/Toggle/Toggle.tsx

```diff
@@ -1,32 +1,54 @@
 import React, { ReactNode, SyntheticEvent, useCallback } from "react";
+import { UseFormRegister } from "react-hook-form";
+
+import { AnyForm } from "types/utility";
 
 import * as TW from "./Toggle.tailwind";
 
 export interface ToggleProps {
   label: ReactNode;
   checked?: boolean;
   onChange?: React.ReactEventHandler<SyntheticEvent>;
+  register?: UseFormRegister<AnyForm>;
   name?: string;
+  title?: string;
 }
 
 export const Toggle: React.FC<ToggleProps> = ({
   label,
   checked = false,
   onChange,
-  name,
+  register,
+  name = "",
+  title,
 }) => {
   // NOTE: the click handlers needs to be on the label to capture all its children clicks and there must always be a <label>
-  const handleClick = useCallback((event) => onChange?.(event), [onChange]);
+  const handleClick = useCallback(
+    (event) => {
+      // default action has to be prevented in order to avoid `double click`
+      onChange && event.preventDefault();
+      onChange?.(event);
+    },
+    [onChange]
+  );
 
   // NOTE: the thumb is a little circle sliding left-right inside a horizontal track
   const thumbClasses = TW[`${!!checked}Thumb`];
   const trackClasses = TW[`${!!checked}Track`];
 
   return (
-    <div className="flex justify-end w-full">
-      <label className="flex items-center cursor-pointer" onClick={handleClick}>
+    <div className="flex flex-col w-full mb-3">
+      {title && (
+        <label className="block text-sm font-medium text-gray-700">
+          {title}
+        </label>
+      )}
+      <label
+        className="flex items-center cursor-pointer mt-3"
+        onClick={handleClick}
+      >
```

Comment:
This might be a start of another long term debug/fix circle with the Toggle component while it already had support for `register` and the code is still there in the `staging` branch. It was already debugged and worked. Thus, there are options:

1. copy the original one and not give it the possibility of `onChange` prop and name it differently
2. copy the original into this one and make some if/else logic that ignores `onChange` if `register` is present
3. start as 1, but have this one copied as well, let's say `ToggleWithHandler`, `ToggleWithRegister` and make `Toggle` just do  

```tsx
   return props.register? <ToggleWithRegister {...props} /> : <ToggleWithHandler {...props} />
``` 


There could be options 4 and 5 and 6... but as this version goes, it looks to me like a footgun.